ui/src/pages/publications/[id]/index.tsx

```diff
@@ -87,31 +96,47 @@ export const getServerSideProps: Types.GetServerSideProps = async (context) => {
     ];
 
     const [
-        publicationVersion,
+        { publicationVersion, versionRequestError },
         bookmarks = [],
         directLinks = { publication: null, linkedTo: [], linkedFrom: [] },
         flags = [],
         topics = []
     ] = await Promise.all(promises);
 
-    if (!publicationVersion) {
+    if (versionRequestError) {
+        const status = versionRequestError.status;
+        if (status === 404 || (token && status === 403)) {
+            return {
+                notFound: true
+            };
+        } else if (status === 403) {
+            return {
+                redirect: {
+                    destination: `${Config.urls.orcidLogin.path}&state=${encodeURIComponent(context.resolvedUrl)}`,
+                    permanent: false
+                }
+            };
+        }
+    }
+
+    if (publicationVersion) {
```

Comment:
Do we need all these if/else checks? I think this part can be simplified like this:

if(!publicationVersion) {
    const status = versionRequestError?.status;
    
    return status === 403 ? { // redirect to login } : { // not found }
}

Reply:
I don't think we can simplify it functionally as we need to deal with 403s in two different ways depending on whether the user is logged in or not (404 if logged in, redirect to login if not).

We can change the if/else on line 108 to use ternary operator, but I don't think it will make much difference. To me, the if/else is slightly easier to read